Update /log endpoint to expect an array

jujaga opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
# Description

<!-- Describe your changes in detail -->
This PR changes the /log endpoint to expect an array of message objects.
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
[SHOWCASE-676](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-676)

## Types of changes

<!-- What types of changes does your code introduce? Uncomment all that apply: -->

<!-- Bug fix (non-breaking change which fixes an issue) -->
New feature (non-breaking change which adds functionality)
<!-- Documentation (non-breaking change with enhancements to documentation) -->
<!-- Breaking change (fix or feature that would cause existing functionality to change) -->

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the OpenAPI 3.0 `v*.api-spec.yaml` documentation (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

jujaga committed:
Update package json eslint definitions
Update default logstash url to use https endpoint

Signed-off-by: Jeremy Ho <jujaga@gmail.com>

jujaga committed:
Rename loggingRouter to logging

Signed-off-by: Jeremy Ho <jujaga@gmail.com>

jujaga committed:
Add 201 response when successful
Use external (temporary) logstash route

Signed-off-by: Jeremy Ho <jujaga@gmail.com>

jujaga committed:
Implement initial array based handling of log entry objects

Signed-off-by: Jeremy Ho <jujaga@gmail.com>

jujaga committed:
Update apitracker to use correct parser
Update validator to expect an array

Signed-off-by: Jeremy Ho <jujaga@gmail.com>

jujaga committed:
Add JSDocs and standardize naming for handling single vs many

Signed-off-by: Jeremy Ho <jujaga@gmail.com>

usingtechnology commented on app/src/components/messageParser.js:
```diff
@@ -15,10 +22,26 @@ const grokParse = (pattern, message) => {
+   *  @returns {object[]} Returns an array of JSON Object for HALPAS logstash
+   */
+  parseMany: async (authorizedParty, obj) => {
+    return await Promise.all(
```
I don't think we want to do a promise all here, we create a timestamp for when get called, the assumption being the caller is sending in data in order.  Timestamps can get out of sync (milliseconds), but data could be represented 'out of order'.

jujaga replied:
We can make that synchronous - and as per discussion down the road consider implementing a log order field or similar. Will look into it on next PR.

usingtechnology reviewed (approved):
looking good!

jujaga committed:
Add explicit eslintignore file
- Inline command wasn't working for some environments
Change default logstashurl location

Signed-off-by: Jeremy Ho <jujaga@gmail.com>
